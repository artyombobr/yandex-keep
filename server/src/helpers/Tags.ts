import {TagsEntity} from '../shared';

class Tags {
  private readonly tags: TagsEntity[];

  private constructor(tags: TagsEntity[]) {
    this.tags = tags;
  }

  public toArray(): TagsEntity[] {
    return [...this.tags];
  }

  public static factory(tags: TagsEntity[]) {
    return new Tags(tags);
  }
}

export default Tags;
