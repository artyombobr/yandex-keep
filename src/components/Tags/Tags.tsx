import * as React from 'react';

function Tags(props: any) {
  const {data, note} = props;
  return(<div className='note__tags'>
      {note.tags.map((tag: any) => {
        return (<div className='note__tag'>{data.tags[tag].tag}</div>)
      })}
    </div>
  )
}

export default Tags;
