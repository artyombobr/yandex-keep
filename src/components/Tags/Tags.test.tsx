import React from 'react';
import { shallow } from 'enzyme';
import Tags from './Tags';

describe('Tags', () => {
  it('correct render', () => {
    const component = shallow(<Tags />);
    expect(component).toMatchSnapshot();
  });
});
