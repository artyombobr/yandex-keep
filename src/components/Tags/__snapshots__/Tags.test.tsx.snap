// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    note={
      Object {
        "color": 0,
        "created": 1551593220000,
        "id": 1,
        "size": "s",
        "tags": Array [
          0,
        ],
        "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
        "type": "text",
      }
    }
    tags={
      Array [
        Object {
          "id": 0,
          "tag": "покупки",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="note__tag"
        >
          покупки
        </div>,
      ],
      "className": "note__tags",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "покупки",
          "className": "note__tag",
        },
        "ref": null,
        "rendered": "покупки",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="note__tag"
          >
            покупки
          </div>,
        ],
        "className": "note__tags",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "покупки",
            "className": "note__tag",
          },
          "ref": null,
          "rendered": "покупки",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
